logstash-core/lib/logstash/pipeline.rb

```diff
@@ -12,20 +12,32 @@
 require "logstash/config/cpu_core_strategy"
 require "logstash/util/defaults_printer"
 require "logstash/shutdown_controller"
+require "logstash/util/wrapped_synchronous_queue"
+require "logstash/pipeline_reporter"
 
 module LogStash; class Pipeline
-  attr_reader :inputs, :filters, :outputs, :input_to_filter, :filter_to_output
+  attr_reader :inputs, :filters, :outputs, :worker_threads, :events_consumed, :events_filtered, :reporter, :pipeline_id
+
+  DEFAULT_SETTINGS = {
+    # We don't set pipeline workers defaults here, but rather in safe_pipeline_workers
```

Comment:
The comment is confusing, I wrote it, it was meant to be read with the PR diff so the reviewer did not ask 'hey is this a mistake, don't we need this setting?